### PR TITLE
Better 404 for courses

### LINKED ISSUE #365: 404 for a /course url should indicate the slug of the nonexistent course
The message should be something like 'no course foo/bar_(baz)' exists.

### PR DESCRIPTION
**Pull request by AmitJoki**
Fixes #365

### CODE REVIEW

**Comment by ragesoss** [comment]
I'd rather find a way to do this without JS.

**Comment by AmitJoki** [comment]
I'll do it in `errors_controller.rb` then :+1:

**Comment by AmitJoki** [comment]
@ragesoss removed javascript and have added it to the controller instead.

File: app/controllers/errors_controller.rb [hunk 1]
```diff
@@ -20,6 +21,16 @@ def incorrect_passcode
     render status: 401
   end
 
+  private
```
Comment (psinghal20) [inline app/controllers/errors_controller.rb]: I don't think the login_error method should be private. That is causing the failing specs in Travis Build. Also, there are some Rubocop linting errors in the code. Please have a look at the Travis build for more info.

Comment (AmitJoki) [reply app/controllers/errors_controller.rb]: yeah. I just pushed the fix.

**Comment by psinghal20** [comment]
@AmitJoki Please update the `error_pages_spec.rb` for the new message that you have added.

File: app/controllers/errors_controller.rb [hunk 2]
```diff
@@ -29,4 +30,14 @@ def login_error
       render status: 200
     end
   end
+
+  private
+
+  def message
```
Comment (ragesoss) [inline app/controllers/errors_controller.rb]: A more specific name would be helpful here, as this is specific to 404 errors.

Comment (AmitJoki) [reply app/controllers/errors_controller.rb]: is `course_not_found` okay?

Comment (ragesoss) [reply app/controllers/errors_controller.rb]: It's also the message for other not found errors, so I'd go with `not_found_message`.

**Comment by ragesoss** [comment]
I don't think it works well in the header. Maybe instead, it could switch the message beneath. It should not say anything about courses if it's *not* a course page, and the smaller-text message beneath should include the slug if it is a course.

Also, it's no longer the case that "If it was recently registered, it may take a little while for the system to add it."

no such course

**Comment by AmitJoki** [comment]
@ragesoss the failing spec is unrelated. Can you please rebuild this?

**Pull request merged**